### PR TITLE
fix: fix async size and conflict handling PE-2258

### PR DESCRIPTION
**Pull request by javdhu**

### CODE REVIEW

File: lib/blocs/upload/upload_cubit.dart [hunk 1]
```diff
@@ -347,12 +349,13 @@ class UploadCubit extends Cubit<UploadState> {
...
   }
 
-  void removeBigFiles() async {
+  Future<void> skipLargeFiles() async {
     for (final file in files) {
       if (await file.ioFile.length > sizeLimit) {
         files.remove(file);
       }
     }
+    checkConflicts();
```
Comment (fedellen) [inline lib/blocs/upload/upload_cubit.dart:358]: Is it okay to not `await` this?

**Review by thiagocarvalhodev — DISMISSED** [review]
LGTM!

File: lib/blocs/upload/upload_cubit.dart (hunk 1 above)
Comment (fedellen) [reply lib/blocs/upload/upload_cubit.dart:358]: Also a naming nit here, `skipLargeFiles` name does not really indicate that we'd also be checking conflicts

**Review by fedellen — DISMISSED** [review]
Left some nits. I suspect the non await will be okay because we are in an `onPressed` event

File: lib/blocs/upload/upload_cubit.dart (hunk 1 above)
Comment (karlprieb) [reply lib/blocs/upload/upload_cubit.dart:358]: If it's the last thing that your function will do it's fine to omit `await` keyword.

**Review by karlprieb — DISMISSED** [review]

**Review by matibat — APPROVED** [review]

**Review by matibat — APPROVED** [review]
LGTM

**Review by thiagocarvalhodev — APPROVED** [review]
LGTM!

**Pull request merged**